jsonrpc/endpoints_debug.go

```diff
@@ -218,7 +218,7 @@ func (d *DebugEndpoints) TraceBatchByNumber(httpRequest *http.Request, number ty
 
 		buffer := make(chan byte, bufferSize)
 
-		mu := sync.Mutex{}
+		mu := &sync.Mutex{}
```

Comment:
This is something I viewed when I started to review the code, why this needs to be a pointer? I think it should work doing the assignement without &, isn't it?

Reply:
I have faced a race condition issue when running the tests and I've found this explanation: https://stackoverflow.com/questions/64735267/race-condition-when-using-mutex-and-reflect

When I've moved the mutex to be a pointer it fixed the problem. Them I decided to review the mutex we had and made them pointers to avoid the problem, since the RPC uses many go routines to handle the requests.